java-engine/src/main/java/io/getunleash/engine/VariantDef.java

```diff
@@ -2,20 +2,24 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Optional;
 
 public class VariantDef {
     private final String name;
     private final Payload payload;
     private final Boolean enabled;
+    private final Boolean featureEnabled;
 
     @JsonCreator
     VariantDef(
             @JsonProperty("name") String name,
             @JsonProperty("payload") Payload payload,
-            @JsonProperty("enabled") Boolean enabled) {
+            @JsonProperty("enabled") Boolean enabled,
+            @JsonProperty("feature_enabled") Boolean featureEnabled) {
         this.name = name;
         this.payload = payload;
         this.enabled = enabled;
+        this.featureEnabled = Optional.ofNullable(featureEnabled).orElse(false);
```

Comment:
I still find it confusing to have enabled and feature enabled as part of the variant def, but I think I'm late to this conversation

Reply:
I don't disagree but I do think you're a bit late to the party. This is already in the client spec and implemented in Python